docs/advanced/extensibleenums.md

```diff
@@ -7,216 +7,139 @@ Extensible Enums are an enhancement of specific Vanilla enums to allow new entri
 
-```java
-// For some enum with codec
-public enum ExampleEnumStringCodec implements StringRepresentable {
-    // ...
-    ;
+#### Retrieving the Generated Constant
 
-    private static final IntFunction<ExampleEnumStringCodec> BY_ID = ByIdMap.continuous(
-        ExampleEnumStringCodec::getId, values(), ByIdMap.OutOfBoundsStrategy.ZERO
-    );
+The generated enum constant can be retrieved via `TheEnum.valueOf(String)`. If a field reference is used to provide the parameters, then the constant can also be retrieved from the `EnumProxy` object via `EnumProxy#getValue()`.
 
-    public static final Codec<ExampleEnumStringCodec> CODEC = StringRepresentable.fromEnum(ExampleEnumStringCodec::values);
-    public static final StreamCodec<ByteBuf, ExampleEnumStringCodec> CODEC = ByteBufCodecs.idMapper(BY_ID, ExampleEnumStringCodec::getId);
+## Contributing to NeoForge
 
-    ExampleEnumStringCodec(int id, String serializedName) {
-        // ...
-    }
+To add a new extensible enum to NeoForge, there are at least two required things to do:
 
-    // ...
-}
+- Make the enum implement `IExtensibleEnum` to mark that this enum should be transformed via the `RuntimeEnumExtender`.
+- Add a `getExtensionInfo` method that returns `ExtensionInfo.nonExtended(TheEnum.class)`.
 
-// The final extended enum may look something like this
-public enum ExampleEnumStringCodec implements StringRepresentable, net.neoforged.neoforge.common.IExtensibleEnum {
-    // ...
-    ;
+Further action is required depending on specific details about the enum:
 
-    private static final java.util.Map<String, ExampleEnumStringCodec> BY_NAME = 
-        java.util.Arrays.stream(ExampleEnumStringCodec.values())
-        .collect(java.util.stream.Collectors.toMap(
-            e -> e.serializedName, e -> e
-        ));
+- If the enum has an int ID parameter which should match the entry's ordinal, then the enum should be annotated with `@NumberedEnum` with the ID's parameter index as the annotation's value if it's not the first parameter
+- If the enum has a String name parameter which is used for serialization and should therefore be namespaced, then the enum should be annotated with `@NamedEnum` with the name's parameter index as the annotation's value if it's not the first parameter
+- If the enum is sent over the network, then it should be annotated with `@NetworkedEnum` with the annotation's parameter specifying in which direction the values may be sent (clientbound, serverbound or bidirectional)
+  - Warning: networked enums will require additional steps once network checks for enums are implemented in NeoForge
+- If the enum has constructors which are not usable by mods (i.e. because they require registry objects on an enum that may be initialized before modded registration runs), then they should be annotated with `@ReservedConstructor`
```

Comment:
This should also be explained above when creating an extensible enum since this means that the constructor cannot be selected.